webui/react/src/types.ts

```diff
@@ -211,25 +211,25 @@ export enum CheckpointState {
   Deleted = 'DELETED',
 }
 
-export interface Checkpoint {
-  endTime? : string;
+interface BaseStep extends StartEndTimes {
   id: number;
+  trialId: number;
+}
+
+export interface Checkpoint extends BaseStep {
```

Comment:
**suggestion (simplify):** I could just be me, but I found the multiple levels of extensions to be much harder to follow. One benefit of this approach is that you can isolate changes to properties of `BaseStep` in a single area, but it also puts us in a tougher place if we want different properties for different interfaces. My recommendation is to just keep the `BaseStep` properties in the respective interfaces.

Reply:
we would have less levels if we could extend multiple things.. actually we might be able to with `&` :thinking:

Reply:
I moved the types closer together (in code location) so they're more easily accessible.

>  but it also puts us in a tougher place if we want different properties for different interfaces

I'm not be missing something from your POV but I think if those properties are truly shared as they are defined here then we shouldn't have this case? :thinking: 

BaseStep now has 4 properties which would be copied across 3 (so far) types if we remove it. I'd lean toward the DRY principle unless we have specific reasons not to : ) let's zoom if you wanna chat more about this.

Reply:
we discussed this and we reduced the nesting by extending multiple types.